ts/main.ts

```diff
@@ -1 +1,44 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
 /* global data */
+
+const $inputURL = document.getElementById('photoURL') as HTMLInputElement;
+const photoPreview = document.getElementById('url-photo') as HTMLImageElement;
+
+if (!$inputURL) throw new Error('inputURL not found');
+if (!photoPreview) throw new Error('photoPreview not found');
+
+function updatePhotoPreview(event: Event): void {
+  const inputElement = event.target as HTMLInputElement;
+  photoPreview.setAttribute('src', inputElement.value);
+}
+
+$inputURL.addEventListener('input', updatePhotoPreview);
+
+/* code above is the event listener for the URL SRC */
+/* code below is the event listener for the submit button */
+
+const entryForm = document.querySelector('#entry-form') as HTMLFormElement;
+const $title = document.querySelector('#title') as HTMLInputElement;
+const $notes = document.querySelector('#notes') as HTMLTextAreaElement;
+
+if (!entryForm) throw new Error('$entryForm query has failed');
+if (!$title) throw new Error('$title query has failed');
+if (!$notes) throw new Error('$notes query has failed');
```

Comment:
While this does work, I would like you to get the value by accessing the FormControlCollection elements like we did in the `typescript-forms` exercise. The problem is that there may be a chance where the values did not update and we have old values. If we wait to access the form and values on submission, we will always get the most updated value from the form.